presto-main/src/main/java/com/facebook/presto/operator/aggregation/AggregationCompiler.java

```diff
@@ -254,4 +301,138 @@ private static Method getCombineFunction(Class<?> clazz, Class<?> stateClass)
 
         return stateClasses;
     }
+
+    private static String getDescription(AnnotatedElement base, AnnotatedElement override)
+    {
+        Description description = override.getAnnotation(Description.class);
+        if (description != null) {
+            return description.value();
+        }
+        description = base.getAnnotation(Description.class);
+        return (description == null) ? null : description.value();
+    }
+
+    private static Set<String> getLiteralParameter(Method inputFunction)
+    {
+        ImmutableSet.Builder<String> literalParametersBuilder = ImmutableSet.builder();
+
+        Annotation[] literalParameters = inputFunction.getAnnotations();
+        for (Annotation annotation : literalParameters) {
+            if (annotation instanceof LiteralParameters) {
+                for (String literal : ((LiteralParameters) annotation).value()) {
+                   literalParametersBuilder.add(literal);
+                }
+            }
+        }
+
+        return literalParametersBuilder.build();
+    }
+
+    public static class BindableAggregationFunction
```

Comment:
I would extract it as a standalone class within the same package and make it's constructor `package-private`